Faeries/refactor yesno question

transfaeries opened the pull request:
Refactores ask_yesno_question to ask a freeform question and parse the answer, as opposed to having it's own functionality. This allows us to remove pending_confirmations, do_yes, do_no. It now allows for people to cancel, which returns None. This might break some flows so we can reconsider that, but I think it's good to allow people to say "actually no, I want out of here, not yes not no"

also now accepts answers that are not just "yes" and "no" like "y" "n" "yeah" etc.

Hopefully removed all the obsoleted codepaths, but please let me know if I missed anything.

technillogue commented on forest/core.py:
```diff
@@ -1355,8 +1339,6 @@ async def ask_address_question(
         if not api:
             logging.error("Error, missing Google Maps API in secrets configuration")
             return None
-        if require_first_device:
-            self.requires_first_device[recipient] = True
```
remove require_first_device as an argument

technillogue replied:
I think it should always be false

technillogue reviewed (commented):
looks good, just need to fix the errors in hotline and mobfriend

transfaeries commented:
all checks passed and removed all the codepaths I could find. since they were in hotline, I would feel better if Ilia looked over it to make sure I didn't break anything for her. Other than that, it should be good.

itdaniher commented:
@transfaeries can you merge in the 'teddy' branch? I refactored yes/no a bit there....

transfaeries commented:
I merged in teddy as requested. Removed obsolete codepaths. Changed some things for the better. added more negative and affirmative answers. Finally fixed MOST of the mypy and pylint errors, but I'm not sure how to fix the rest

itdaniher reviewed (approved):
thrilling! looks good to me, to squash-merge. Thanks for nabbing the synonyms from the other branch! This directly addresses feedback from last night's event, so I'll make sure to merge into teddy.

Cheers!

itdaniher commented:
this is a big UX win... will be trying to move hotline etc over in the immediate future!
